File: src/main/kotlin/de/gieskerb/class_craft/data/HorseData.java
package de.gieskerb.class_craft.data;

import org.bukkit.Material;
import org.bukkit.entity.Horse;
import org.bukkit.entity.Player;
import org.json.simple.JSONObject;

public class HorseData  {

    public Horse.Color color;
    public Horse.Style style;
    public double speed, jump, health;
    public String name;
    public Material armor;

    public HorseData(Horse.Color color, Horse.Style style, double speed, double jump, double health, String name, Material armor) {
        this.color = color;
        this.style = style;
        this.speed = speed;
        this.jump = jump;
        this.health = health;
        this.name = name;
        this.armor = armor;
    }

    public HorseData(JSONObject json) {
        this.color = Horse.Color.values()[Integer.parseInt(json.get("color").toString())];
        this.style = Horse.Style.values()[Integer.parseInt( json.get("style").toString())];
        this.speed = Double.parseDouble(json.get("speed").toString());
        this.jump = Double.parseDouble(json.get("jump").toString());
        this.health = Double.parseDouble(json.get("health").toString());
        this.name = json.get("name").toString();
        this.armor = Material.valueOf(json.get("armor").toString());
    }

    public static String getHorseName(Player player, String className) {
        String playerName = player.getName();
        if (playerName.charAt(playerName.length() - 1) == 's' || playerName.charAt(playerName.length() - 1) == 'x' ||
                playerName.charAt(playerName.length() - 1) == 'ß') {
            playerName += '\'';
        } else {
            playerName += 's';
        }
        return playerName + " " + className + "-Horse";
    }

    public static HorseData fromJSON(JSONObject json) {
        if (json.isEmpty()) return null;
        return new HorseData(json);
    }


    public JSONObject toJSON() {
        JSONObject json = new JSONObject();
        json.put("color", color.ordinal());
        json.put("style", style.ordinal());
        json.put("speed", speed);
        json.put("jump", jump);
        json.put("health", health);
        json.put("name", name);
        json.put("armor", armor.toString());
        return json;
    }
}

File: src/main/kotlin/de/gieskerb/class_craft/data/HorseData.kt
package de.gieskerb.class_craft.data

import org.bukkit.Material
import org.bukkit.entity.Horse
import org.bukkit.entity.Player
import org.json.simple.JSONObject

class HorseData {
    var color: Horse.Color
    var style: Horse.Style
    val speed: Double
    val jump: Double
    val health: Double
    val name: String
    val armor: Material

    constructor(
        color: Horse.Color,
        style: Horse.Style,
        speed: Double,
        jump: Double,
        health: Double,
        name: String,
        armor: Material
    ) {
        this.color = color
        this.style = style
        this.speed = speed
        this.jump = jump
        this.health = health
        this.name = name
        this.armor = armor
    }

    private constructor(json: JSONObject) {
        this.color = Horse.Color.entries.toTypedArray()[json["color"].toString().toInt()]
        this.style = Horse.Style.entries.toTypedArray()[json["style"].toString().toInt()]
        this.speed = json["speed"].toString().toDouble()
        this.jump = json["jump"].toString().toDouble()
        this.health = json["health"].toString().toDouble()
        this.name = json["name"].toString()
        this.armor = Material.valueOf(json["armor"].toString())
    }

    fun toJSON(): JSONObject {
        val json = JSONObject()
        json["color"] = color.ordinal
        json["style"] = style.ordinal
        json["speed"] = speed
        json["jump"] = jump
        json["health"] = health
        json["name"] = name
        json["armor"] = armor.toString()
        return json
    }

    companion object {
        fun horseNameByPlayer(player: Player, className: String): String {
            var playerName = player.name
            playerName += if (playerName[playerName.length - 1] == 's' || playerName[playerName.length - 1] == 'x' || playerName[playerName.length - 1] == 'ß') {
                '\''
            } else {
                's'
            }
            return "$playerName $className-Horse"
        }

        fun fromJSON(json: JSONObject): HorseData? {
            if (json.isEmpty()) return null
            return HorseData(json)
        }
    }
}